### ISSUE TITLE
Building for wasm doesn't work (well, building works but running does not)

### ISSUE THREAD

**Issue by apoelstra** [rust-bitcoin/rust-secp256k1#124]
Related to #119 

We currently depend on the following symbols:
* `malloc`
* `free`

(We additionally depended on `abort` and `printf` until #115 )

This prevents use with rust-wasm for this library and all downstream users (e.g. rust-bitcoin). While doing any cryptography in a browser is [very ill-advised](https://www.nccgroup.trust/us/about-us/newsroom-and-events/blog/2011/august/javascript-cryptography-considered-harmful/), rust-secp does a lot more than just signing and verification (it can also parse and display keys), and its downstream deps such as rust-bitcoin do many things that certainly should be possible and easy to do with rust-wasm.

We also don't want to encourage users to find other pure-Rust secp256k1 libraries with dramatically less review and QA on them, and to use these in a Bitcoin context.

See https://github.com/rust-bitcoin/rust-bitcoin/pull/240 for earlier discussion.

**Comment by apoelstra** [comment]
I was able to get a wasm application working which parses and Javascript `alert`s a public key. Unfortunately to do so I needed to

1. Hack #116 to eliminate all calls to `context_create` and `context_destroy` (this I expected).
2. Actually delete `context_create`, `scratch_space_create`, `context_destroy`, `scratch_space_destroy` from the libsecp source. Deleting the FFI bindings was not enough (this I did not expect).
